rust/core/src/driver_manager.rs

```diff
@@ -0,0 +1,1218 @@
+        check_status(status, error)?;
+        let reader = ArrowArrayStreamReader::try_new(stream)?;
+        Ok(reader)
+    }
+
+    fn execute_schema(&mut self) -> Result<arrow::datatypes::Schema> {
+        let driver = &self.inner.connection.database.driver.driver.lock().unwrap();
+        let mut statement = self.inner.statement.lock().unwrap();
+        let mut error = ffi::FFI_AdbcError::with_driver(driver);
+        let method = driver_method!(driver, StatementExecuteSchema);
+        let mut schema = FFI_ArrowSchema::empty();
+        let status = unsafe { method(statement.deref_mut(), &mut schema, &mut error) };
+        check_status(status, error)?;
+        Ok((&schema).try_into()?)
+    }
+
+    fn execute_update(&mut self) -> Result<Option<i64>> {
+        let driver = &self.inner.connection.database.driver.driver.lock().unwrap();
+        let mut statement = self.inner.statement.lock().unwrap();
+        let mut error = ffi::FFI_AdbcError::with_driver(driver);
+        let method = driver_method!(driver, StatementExecuteQuery);
+        let mut rows_affected: i64 = -1;
+        let status = unsafe {
+            method(
+                statement.deref_mut(),
+                null_mut(),
+                &mut rows_affected,
+                &mut error,
+            )
+        };
+        check_status(status, error)?;
+        let rows_affected = if rows_affected == -1 {
+            None
+        } else {
+            Some(rows_affected)
+        };
+        Ok(rows_affected)
+    }
+
+    fn execute_partitions(&mut self) -> Result<PartitionedResult> {
+        let driver = &self.inner.connection.database.driver.driver.lock().unwrap();
+        let mut statement = self.inner.statement.lock().unwrap();
+        let mut error = ffi::FFI_AdbcError::with_driver(driver);
+        let method = driver_method!(driver, StatementExecutePartitions);
+        let mut schema = FFI_ArrowSchema::empty();
+        let mut partitions = ffi::FFI_AdbcPartitions::default();
+        let mut rows_affected: i64 = -1;
+        let status = unsafe {
+            method(
+                statement.deref_mut(),
+                &mut schema,
+                &mut partitions,
+                &mut rows_affected,
+                &mut error,
+            )
+        };
+        check_status(status, error)?;
+
+        let result = PartitionedResult {
+            partitions: partitions.into(),
+            schema: (&schema).try_into()?,
+            rows_affected,
+        };
+
+        Ok(result)
+    }
+
+    fn get_parameters_schema(&self) -> Result<arrow::datatypes::Schema> {
+        let driver = &self.inner.connection.database.driver.driver.lock().unwrap();
+        let mut statement = self.inner.statement.lock().unwrap();
+        let mut error = ffi::FFI_AdbcError::with_driver(driver);
+        let method = driver_method!(driver, StatementGetParameterSchema);
```

Comment:
Just noticing now that this method uses plural (`parameters`) vs singular (`Parameter`) for the method.